lib/rucio/core/did.py

```diff
@@ -2734,6 +2734,7 @@ def set_status(scope, name, session=None, **kwargs):
     :param kwargs:  Keyword arguments of the form status_name=value.
     """
     statuses = ['open', ]
+    reevaluate_close_did = config_get_bool('core', 'reevaluate_dids_at_close', raise_exception=False, default=False, session=session)
```

Comment:
Do we really need this? I think less configuration is better, so if really needed, I would make this per default `True`, but perhaps we can live without it?

Reply:
Not sure it should be `True` by default. Different communities will have different expectations. I don't know how many communities rely on subscriptions, but if we always reevaluate it can cause unwanted replications.

Reply:
Does it?
I thought one of the things we need to make sure is that this doesn't happen. Thus the transmogrifier checks to not create rules twice. If this is the case, I would not offer this option (Or at least make it `True` by default).
If this is not the case, then we should look into this. But assuming things work correctly, this shouldn't lead to any side-effects?

Reply:
One can have also something like : A dataset is evaluated, but after some time one wants to decrease the number of copies before it's closed. Then when the dataset is closed, a new copy is made. Also, in the case of `split_rule` if a RSE is initially chosen that matches the RSE expression, but shortly after some RSE attribute is changed and then this RSE does not match the RSE expression anymore. In that case, the transmogrifier will create another replica on a new RSE.

Reply:
Yes you are right, I think in this case there is no way to detect it. Ok, let's make it opt-in then. 
I would suggest a different name though and adding it to https://rucio.cern.ch/documentation/configuration_parameters as well.

I would suggest
```
[subscriptions]
reevaluate_dids_at_close
```
So same name, just different category.

Reply:
Done